src/core/eme/attach_media_keys.ts

```diff
@@ -25,11 +25,43 @@ import log from "../../log";
 import MediaKeysInfosStore from "./media_keys_infos_store";
 import { IMediaKeysInfos } from "./types";
 
+/**
+ * Close sessions on loaded sessions store if it has changed.
+ * Dispose the old media keys if different than new one.
+ * @param {Object} mediaKeysInfos
+ * @param {Object} mediaElement
+ * @returns {Observable}
+ */
+export function cleanMediaKeys(
+  mediaKeysInfos: IMediaKeysInfos,
```

Comment:
For a function called `cleanMediaKeys`, I find it weird to need the MediaKeys to which it should not be set to. Or any `mediaKeysInfo`, really.

Either it is a bad name or we should check if the new MediaKeys is different than the old one elsewhere.

Reply:
To me it still isn't clear from the name that this function only remove the current mediaKeys in some conditions.

Also, because both `disableOldMediaKeys` and `attachMediaKeys` check the new MediaKeys against the current state, I would argue that this logic should be moved higher up.